### PR TITLE
[impeller] support generating mip-maps on Vulkan

### PR DESCRIPTION
**Pull request by iskakaushik**
fix: https://github.com/flutter/flutter/issues/120134

### CODE REVIEW

**Comment by iskakaushik** [comment]
image

File: impeller/renderer/backend/vulkan/allocator_vk.cc [hunk 1]
```diff
@@ -116,6 +116,7 @@ std::shared_ptr<Texture> AllocatorVK::OnCreateTexture(
   image_create_info.initialLayout = vk::ImageLayout::eUndefined;
   image_create_info.usage = vk::ImageUsageFlagBits::eSampled |
```
Comment (chinmaygarde) [inline impeller/renderer/backend/vulkan/allocator_vk.cc:117]: This makes me think we should perhaps have a recommended usage as on optional on the texture descriptor. For later though.

File: shell/platform/android/platform_view_android.cc [hunk 2]
```diff
@@ -47,7 +47,7 @@ std::unique_ptr<AndroidSurface> AndroidSurfaceFactoryImpl::CreateSurface() {
     case AndroidRenderingAPI::kOpenGLES:
       if (enable_impeller_) {
 // TODO(kaushikiska@): Enable this after wiring a preference for Vulkan backend.
-#if false
+#if true
```
Comment (chinmaygarde) [inline shell/platform/android/platform_view_android.cc:50]: Did you mean to leave this in?

**Review by chinmaygarde — APPROVED** [review]
LGTM but I think you forgot to back out the part where you opted into Vulkan on Android.

File: shell/platform/android/platform_view_android.cc (hunk 2 above)
Comment (iskakaushik) [reply shell/platform/android/platform_view_android.cc:50]: nope, typo!

**Pull request merged**